target/bin/addalias

```diff
@@ -5,7 +5,7 @@ source /usr/local/bin/helpers/index.sh
 
 function _main
 {
-  [[ ${1:-} == 'help' ]] && { __usage ; exit 0 ; }
+  [[ ${#} -lt 2 ]] && { __usage ; exit 0 ; }
```

Comment:
Would it make sense to be DRY here and just have these all use a helper?

`addalias` and `adddovecotmasteruser` both had the same content originally, but you've diverged it due to number of args? If we had a method that just set the `-lt` value to compare to, would that work for all cases being changed here?

Reply:
Hm, I had the same idea. But then I dismissed the thought.

For the previous `  [[ ${1:-} == 'help' ]] && { __usage ; exit 0 ; }` that would have been the same. We haven't used a function here also.

I think this is so small and flat - not worth it. And who knows, maybe there are some cases in the future, where some files needs more flexibility than a standard function can provide?

After all, I am fine to change that to use a function - if the majority want's that 😎

Reply:
> We haven't used a function here also.

That is because before I did a refactor pass, it was a mess and I wanted to minimize the already large diff for that PR I think :sweat_smile: 

> I think this is so small and flat - not worth it. And who knows, maybe there are some cases in the future, where some files needs more flexibility than a standard function can provide?

I believe the `delmailuser` already has something in place to check for `-y`.

Up to you, I just noticed you're introducing two variants now, and one of those becomes multiple lines, while the other uses new syntax (_I roughly could guess what it was, but still looked it up to verify_).

Personally for maintaining things like this, I prefer the DRY approach suggested, even though it's fairly small, I've seen contributions in the past for these files make small changes to a subset of files (or just one file) with the shared duplicate code, and then we get this divergence that's caused some bugs. It's not likely to happen with current maintainers, but it has been an issue before.

> After all, I am fine to change that to use a function - if the majority want's that

I'm fine either way. Ideally if I had the time, I'd rather port this all to something like rust :sweat_smile:

Reply:
Check this commit: [`4f2c92c`](https://github.com/docker-mailserver/docker-mailserver/pull/2854/commits/4f2c92c0925df46781e0795b1009eb58bbe5cccf)